app/src/main/java/tech/relaycorp/ping/domain/ReceivePong.kt

```diff
@@ -0,0 +1,40 @@
+package tech.relaycorp.ping.domain
+
+import kotlinx.coroutines.flow.first
+import tech.relaycorp.awaladroid.messaging.IncomingMessage
+import tech.relaycorp.ping.awala.AwalaPing
+import tech.relaycorp.ping.awala.PingSerialization
+import tech.relaycorp.ping.common.Logging.logger
+import tech.relaycorp.ping.data.database.dao.PingDao
+import java.time.ZonedDateTime
+import java.util.logging.Level
+import javax.inject.Inject
+
+class ReceivePong
+@Inject constructor(
+    private val pingDao: PingDao,
+    private val pingSerialization: PingSerialization
+) {
+
+    suspend fun receive(incomingMessage: IncomingMessage) {
+        if (incomingMessage.type != AwalaPing.V1.PongType) {
+            incomingMessage.ack()
```

Comment:
This should also return to skip the rest of the function

Reply:
Silly me.